### PR TITLE
FW-589

### PR DESCRIPTION
**Pull request by Dr-Slugma**
Should wait for FW-879 to be merged before merging into master as this branch contains some changes from FW-879.

### CODE REVIEW

**Comment by dy2288** [comment]
Can one of the admins verify this patch?

File: frontend/app/assets/javascripts/views/pages/explore/dialect/play/hangman/index.js [hunk 1]
```diff
@@ -130,6 +134,14 @@ export class Hangman extends Component {
       this.props.routeParams.dialect_path + '/Dictionary'
     )
 
+    if (selectn('response.resultsCount', computeWords) < MIN_REQ_WORDS) {
```
Comment (coakenfold) [inline frontend/app/assets/javascripts/views/pages/explore/dialect/play/hangman/index.js]: Wondering if you want `<=` here?
`if (5 < 5)`
vs
`if (5 <= 5)`

File: frontend/app/assets/javascripts/views/pages/explore/dialect/play/hangman/index.js [hunk 2]
```diff
@@ -149,13 +161,22 @@ export class Hangman extends Component {
...
+        let temp = array[i]
+        array[i] = array[rand]
+        array[rand] = temp
+      }
+    }
 
     if (word_array.length > 0) {
       //Since the alphabet isn't complete, we need fill in the rest
       const character_string = word_array.map((word) => word.puzzle).join('')
       const unique_characters = Array.from(new Set(character_string.split(/(?!$)/u))).filter((v) => v != ' ')
```
Comment (coakenfold) [inline frontend/app/assets/javascripts/views/pages/explore/dialect/play/hangman/index.js]: Realize you didn't update this bit of code but want to flag it as a possible source for splitting grapheme clusters, ie: https://firstvoices.atlassian.net/browse/FW-705

> splitting a string along code units will fail with Grapheme Clusters.
> So, given foo̅, if you split it by code units you'll end up with [ "f", "o", "o", "̅" ].

File: frontend/scripts/files/testLangSevenAlphabet.csv [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+$',,,36,,,,,
+$&,,,37,,,,,
+$`,,,38,,,,,
+$1,,,39,,,,,
+g,,,40,,,,,
+i,,,41,,,,,
+m,,,42,,,,,
+u,,,43,,,,,
+y,,,44,,,,,
+:,,,45,,,,,
```
Comment (coakenfold) [inline frontend/scripts/files/testLangSevenAlphabet.csv]: Should add some Grapheme Clusters to this word list, eg: ẕ, o̅, etc.

Comment (coakenfold) [reply frontend/scripts/files/testLangSevenAlphabet.csv]: Oh, didn't notice this was for an alphabet. May still be relevant though.

File: frontend/scripts/files/testLangSevenWord.csv [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+m,noun,TestPronunciation,TestTranslation,,TestLiteralTranslation,TestCategory,TestCulturalNote,,TestReference,TestAcknowledgement,1,1,TestWordAudio,TestAudioDescription,test_related_audio.wav,0,0,,,,,,,,,,,,,,,,,TESTLANGUAGESEVEN_ADMIN,,,,,,,,,,,,
+u,noun,TestPronunciation,TestTranslation,,TestLiteralTranslation,TestCategory,TestCulturalNote,,TestReference,TestAcknowledgement,1,1,TestWordAudio,TestAudioDescription,test_related_audio.wav,0,0,,,,,,,,,,,,,,,,,TESTLANGUAGESEVEN_ADMIN,,,,,,,,,,,,
+y,noun,TestPronunciation,TestTranslation,,TestLiteralTranslation,TestCategory,TestCulturalNote,,TestReference,TestAcknowledgement,1,1,TestWordAudio,TestAudioDescription,test_related_audio.wav,0,0,,,,,,,,,,,,,,,,,TESTLANGUAGESEVEN_ADMIN,,,,,,,,,,,,
+:,noun,TestPronunciation,TestTranslation,,TestLiteralTranslation,TestCategory,TestCulturalNote,,TestReference,TestAcknowledgement,1,1,TestWordAudio,TestAudioDescription,test_related_audio.wav,0,0,,,,,,,,,,,,,,,,,TESTLANGUAGESEVEN_ADMIN,,,,,,,,,,,,
```
Comment (coakenfold) [inline frontend/scripts/files/testLangSevenWord.csv]: Should add some Grapheme Clusters to this word list, eg: ẕ, o̅, etc.

Comment (Dr-Slugma) [reply frontend/scripts/files/testLangSevenWord.csv]: I'll add a few Grapheme clusters to the Alphabet and the testLang.

File: frontend/app/assets/javascripts/views/pages/explore/dialect/play/hangman/index.js [hunk 5]
```diff
@@ -130,6 +134,14 @@ export class Hangman extends Component {
       this.props.routeParams.dialect_path + '/Dictionary'
     )
 
+    if (selectn('response.resultsCount', computeWords) < MIN_REQ_WORDS) {
+      return (
+        <div>
+          Game not available: At least 5 child-friendly words with audio are required for this game... Found{' '}
```
Comment (coakenfold) [inline frontend/app/assets/javascripts/views/pages/explore/dialect/play/hangman/index.js]: This line makes me think you want to test `5 <= 5`

If not, shouldn't it read: "Game not available: At least 4 child-friendly words..."?

Comment (coakenfold) [reply frontend/app/assets/javascripts/views/pages/explore/dialect/play/hangman/index.js]: Thanks for clarifying the logic. I misinterpreted it.

**Comment by coakenfold** [comment]
Removed the change request due to a misunderstanding on my part. Good to go.

**Pull request merged**